go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -516,11 +517,57 @@ func ConfigureServices(
 	controller.Register(InitMetricsListener)
 
 	InitLockSuperUser := &svcs.AnonService{
-		InitF: func(context.Context) error {
+		InitF: func(ctx context.Context) error {
 			mysqlDb := sqlEngine.GetUnderlyingEngine().Analyzer.Catalog.MySQLDb
+
+			host := "localhost"
+			// LocalConnectionUser is reserved for `dolt sql` otherwise the ephemeral user can't be created
+			reserved := [...]string{LocalConnectionUser} // users necessary at startup to enter dolt sql-server
+
+			// check for reserved user conflicts
+			rd := mysqlDb.Reader()
+			var warn strings.Builder
+			conflicts := make([]mysql_db.UserPrimaryKey, 0, len(reserved))
+			for _, user := range reserved {
+				conflict, _ := rd.GetUser(mysql_db.UserPrimaryKey{Host: host, User: user})
+				if conflict != nil && !conflict.IsEphemeral {
+					warn.WriteString(fmt.Sprintf("Dropped persistent '%s@%s' as it conflicts with dolt reserved '%s@%s'", conflict.User, conflict.Host, user, host))
```

Comment:
(minor) Instead of repeating the username@host (they'll always be the same, right), how about an error message like:
`Dropping persisted user '__dolt_local_user__@localhost' because this account name is reserved for Dolt`